Firestore/Source/API/FIRFieldPath.mm

```diff
@@ -16,34 +16,55 @@
 
 #import "Firestore/Source/API/FIRFieldPath+Internal.h"
 
-#import "Firestore/Source/Model/FSTPath.h"
+#include <functional>
+#include <string>
+#include <utility>
+#include <vector>
+
 #import "Firestore/Source/Util/FSTUsageValidation.h"
 
+#include "Firestore/core/src/firebase/firestore/model/field_path.h"
+#include "Firestore/core/src/firebase/firestore/util/string_apple.h"
+
+namespace util = firebase::firestore::util;
+using firebase::firestore::model::FieldPath;
+
 NS_ASSUME_NONNULL_BEGIN
 
+@interface FIRFieldPath () {
+  /** Internal field path representation */
+  firebase::firestore::model::FieldPath _internalValue;
+}
+
+@end
+
 @implementation FIRFieldPath
 
 - (instancetype)initWithFields:(NSArray<NSString *> *)fieldNames {
   if (fieldNames.count == 0) {
     FSTThrowInvalidArgument(@"Invalid field path. Provided names must not be empty.");
   }
 
+  std::vector<std::string> field_names{};
```

Comment:
Rather than this pattern of creating and then resizing, I believe you can just do this:
```
vector<string> field_names(fieldNames.count);
```

Reply:
I'll use emplace_back to add element at the end of vector. So keep this vector empty instead of `count` empty strings.